test/elixir_analyzer/exercise_test/common_checks/module_pascal_case_test.exs

```diff
@@ -0,0 +1,100 @@
+# credo:disable-for-this-file Credo.Check.Readability.ModuleNames
+
+defmodule ElixirAnalyzer.ExerciseTest.CommonChecks.ModulePascalCaseTest do
+  use ExUnit.Case
+  alias ElixirAnalyzer.Comment
+  alias ElixirAnalyzer.Constants
+  alias ElixirAnalyzer.ExerciseTest.CommonChecks.ModulePascalCase
+
+  test "returns empty list if there are no module definition" do
+    code =
+      quote do
+        def calculate(1), do: 1
+
+        def calculate(n) do
+          n * calculate(n - 1)
+        end
+      end
+
+    assert ModulePascalCase.run(code) == []
+  end
+
+  test "returns empty list if there are no module names not in PascalCase" do
+    code =
+      quote do
+        defmodule Factorial do
+          defmodule FactorialSubmodule do
+            defstruct :!
+          end
+        end
+      end
+
+    assert ModulePascalCase.run(code) == []
+  end
+
+  test "returns an actionable comment with params" do
+    code =
+      quote do
+        defmodule factorial do
+        end
+      end
+
+    assert ModulePascalCase.run(code) == [
+             {:fail,
+              %Comment{
+                type: :actionable,
+                comment: Constants.solution_module_pascal_case(),
+                params: %{
+                  expected: "Factorial",
+                  actual: "factorial"
+                }
+              }}
+           ]
+  end
+
+  test "only reports the first module name" do
+    code =
+      quote do
+        defmodule factorial do
+          defmodule factorial_submodule do
+            defstruct :!
+          end
+        end
+      end
+
+    assert ModulePascalCase.run(code) == [
+             {:fail,
+              %Comment{
+                type: :actionable,
+                comment: Constants.solution_module_pascal_case(),
+                params: %{
+                  expected: "Factorial",
+                  actual: "factorial"
+                }
+              }}
+           ]
+  end
+
+  test "reports the inner module name" do
+    code =
+      quote do
+        defmodule Factorial do
+          defmodule factorial_submodule do
+            defstruct :!
+          end
+        end
+      end
+
+    assert ModulePascalCase.run(code) == [
+             {:fail,
+              %Comment{
+                type: :actionable,
+                comment: Constants.solution_module_pascal_case(),
+                params: %{
+                  expected: "FactorialSubmodule",
+                  actual: "factorial_submodule"
+                }
+              }}
+           ]
+  end
```

Comment:
On second thought... let's add some more tests for corner cases (that fail atm):
```suggestion
  end
  
  test "reports a module name with many segments" do
    code =
      quote do
        defmodule MyLibrary.Math_ops.Factorial do
        end
      end

    assert ModulePascalCase.run(code) == [
             {:fail,
               %Comment{
                 type: :actionable,
                 comment: Constants.solution_module_pascal_case(),
                 params: %{
                   expected: "MyLibrary.MathOps.Factorial",
                   actual: "MyLibrary.Math_ops.Factorial"
                 }
               }}
           ]
  end

  test "reports an inner module name with many segments" do
    code =
      quote do
        defmodule MyLibrary do
          defmodule Math_ops.Factorial do
          end
        end
      end

    assert ModulePascalCase.run(code) == [
             {:fail,
               %Comment{
                 type: :actionable,
                 comment: Constants.solution_module_pascal_case(),
                 params: %{
                   expected: "MathOps.Factorial",
                   actual: "Math_ops.Factorial"
                 }
               }}
           ]
  end
```

Reply:
Thank you, I was careless.

I did not consider segmented module names AND I did not realize that module names cannot start with a lowercase character to begin with, so many of my tests weren't valid. 

It would have been found out if the `imperfect_solution`'s tests had been ran as well. Is there a way to add that test?